File: NNPRO-employee-management-main/NNPRO-employee-management-main/implementation/backend/src/main/java/upce/backend/entity/Attendance.java
package upce.backend.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
public class Attendance {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private LocalDateTime startTime;

    private LocalDateTime endTime;

    private String workDescription;

    @ManyToOne
    @JoinColumn(name = "task_id", nullable = false)
    private Task task;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private AppUser user;
}

File: NNPRO-employee-management-main/NNPRO-employee-management-main/implementation/frontend/src/types/attendance.ts
import {PaginatedResponse} from "./pagination.ts";
import {Member} from "./member.ts";
import {User} from "./auth.ts";

export interface Attendance {
    id: number;
    startTime: string;
    endTime: string;
    workDescription: string;
    taskId: number;
    appUser: Member | User;
}

export interface AttendanceResponse {
    success: boolean;
    attendance?: Attendance;
    message?: string;
}

export interface AttendancesResponse {
    success: boolean;
    attendances?: Attendance[];
    message?: string;
    pagination?: PaginatedResponse;
}